contracts/GoodGhosting.sol

```diff
@@ -256,9 +256,8 @@ contract GoodGhosting is Ownable, Pausable {
         require(!redeemed, "Redeem operation already happened for the game");
         redeemed = true;
         // aave has 1:1 peg for tokens and atokens
-        uint256 adaiBalance = AToken(adaiToken).balanceOf(address(this));
         // there is no redeem function in v2 it is replaced by withdraw in v2
-        lendingPool.withdraw(address(daiToken), adaiBalance, address(this));
+        lendingPool.withdraw(address(daiToken), uint(-1), address(this));
```

Comment:
is this correct?

Reply:
yes this will just withdraw the max adai balance there is in the contract this seems more logical

Reply:
@viraj124 
well, it makes sense to withdraw everything from aave. What I didn't understand is why we're casting `-1` as a `uint` (unsigned integer, which goes from 0 up to 2**32 - 1), which doesn't cover `-1`.

Reply:
Yeah @fodisi this is acceptable uint(-1) corresponds to the max uint which is 2^256 -1 also check https://docs.aave.com/developers/the-core-protocol/lendingpool#withdraw Aave recommends the same

Reply:
OK. I see that Aave recommends passing ([uint(-1)](https://docs.aave.com/developers/the-core-protocol/lendingpool#withdraw)). I found this [solidity issue](https://github.com/ethereum/solidity/issues/533) that actually explains how/why this works.